src/main/java/bio/terra/cli/businessobject/resources/GcsBucket.java

```diff
@@ -100,6 +94,27 @@ public static GcsBucket createControlled(CreateUpdateGcsBucket createParams) {
     return new GcsBucket(createdResource);
   }
 
+  /** Update a GCS bucket resource in the workspace. */
+  public void update(UpdateGcsBucketParams updateParams) {
+    if (updateParams.resourceFields.name != null) {
+      validateEnvironmentVariableName(updateParams.resourceFields.name);
+    }
+    switch (stewardshipType) {
+      case REFERENCED:
+        WorkspaceManagerService.fromContext()
+            .updateReferencedGcsBucket(
+                Context.requireWorkspace().getId(), id, updateParams.resourceFields);
```

Comment:
this is a little funny in that it throws away a lot of the `UpdateGcsBucketParams` fields. Should we just have 2 methods, one for referenced & one for controlled resources update rather than pretending they take the same arguments?

Reply:
that's fair. I originally had 2 methods, but thought it was better if the command classes didn't have to do different things depending on the stewardship type. On second thought, the command classes have to do argument validation differently based on stewardship type anyway, so this isn't really encapsulating anything. will change back to 2 separate methods.